core/src/validator.rs

```diff
@@ -1295,7 +1295,7 @@ fn new_banks_from_ledger(
                 blockstore.clone(),
                 exit,
                 enable_rpc_transaction_history,
-                config.rpc_config.enable_cpi_and_log_storage,
+                config.rpc_config.enable_cpi_and_log_storage || transaction_notifier.is_some(),
```

Comment:
I'm not sure this is intuitive. If `enable_cpi_and_log_storage` is really required when plugins are loaded, we should use a `clap::Arg::requires` to force it visibly rather than silently upgrade like this.  If it's _not_ strictly required, but rather dependent on what a particular plugin needs, then I think we need to find a better way to communicate that.  The concern here is that RPC TX History service is not zero-cost and it's very likely that an accounts db plugin user is looking for high-performance

Reply:
And this flag only used in https://github.com/solana-labs/solana/blob/e3b137066d89c303ad75a1073b90b9e2e32fd23d/ledger/src/blockstore_processor.rs#L1434 when deciding if to take out the logs and inner instructions before send.

Reply:
To follow up on this -- I think it makes more sense to move the logic of if to save the logs into db in the receiver side in TransactionStatusService instead of TransactionStatusSender. So that in the following case
1. Transaction data is streamed via plugin and 2. rpc history service is also enabled, yet when enable_cpi_and_log_storage off, we can support stream the logs to plugin without impacting rpc side. @t-nelson  does it make sense?

Reply:
Can we just plumb `config.rpc_config.enable_cpi_and_log_storage` through to where the plugin is loaded and exit with an error message saying that it's required when the plugin requires it?

Reply:
Using one config.rpc_config.enable_cpi_and_log_storage has a problem that it cannot satisfy different needs between regular RPC and the plugin. E.g in regular RPC -- the user does not wish for the logs being saved while in the plugin side it requires the logs to be saved. I think we probably need explicit flag like 'enable_plugin_cpi_and_log_storage' to decouple them.